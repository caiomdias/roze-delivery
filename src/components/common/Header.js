import React from 'react';
import { useHistory, useLocation } from 'react-router-dom'
import Search from '../common/Search';
import MiniCart from '../common/MiniCart';

const Header = (props) => {
	const history = useHistory();
	const { pathname } = useLocation();
	const { config } = props;

	return (
		<header className="header">
			<div className="container content">
				<div onClick={() => history.push('/')} className="header__logo">
					<img style={config.style} src={config.image} alt="Jose Entregas"/>
				</div>
				{
					config.hasSearch && (
						<div className="header__search">
							<Search
								type="products"
								placeholder="Pesquise sua bebida favorita"
							/>
						</div>
					)
				}
				{ pathname !== '/' && ( <MiniCart /> )}
			</div>
		</header>
	)
};

export default Header;
