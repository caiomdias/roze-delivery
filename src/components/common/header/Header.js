import React from 'react';
import { useHistory, useLocation } from 'react-router-dom'
import Search from '../search/Search';
import MiniCart from '../miniCart/MiniCart';

const Header = (props) => {
	const history = useHistory();
	const { pathname } = useLocation();
	const { config } = props;

	return (
		<header className="header">
			<div className="container content">
				<div onClick={() => history.push('/')} className="header__logo">
					<img style={config && config.style} src={config && config.image} alt="Jose Entregas"/>
				</div>
				{/* {
					config.hasSearch && (
						<div className="header__search">
							<Search
								type="products"
								placeholder="Pesquise sua bebida favorita"
							/>
						</div>
					)
				} */}
				{/* { pathname !== '/' && ( <MiniCart /> )} */}
			</div>
		</header>
	)
};

export default Header;
