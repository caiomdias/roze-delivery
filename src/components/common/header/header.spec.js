import React from 'react';
import renderer from 'react-test-renderer';
import Header from './Header';

jest.mock('react-router-dom', () => ({
    useLocation: jest.fn().mockReturnValue({
      pathname: '/',
      search: '',
      hash: '',
      state: null,
      key: '5nvxpbdafa',
	}),
	useHistory: jest.fn().mockReturnValue({
		push: jest.fn().mockReturnValue()
	})
}));

const config = {
	image: '',
	style: ''
}

describe("Header Component", () => {
	it('render correctly Header component', () => {  
		const headerComponent = renderer.create(<Header config={config}/>);
		expect(headerComponent).toMatchSnapshot();
	});
});