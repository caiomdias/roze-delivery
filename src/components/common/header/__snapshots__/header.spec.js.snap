// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Header Component render correctly Header component 1`] = `
<header
  className="header"
>
  <div
    className="container content"
  >
    <div
      className="header__logo"
      onClick={[Function]}
    >
      <img
        alt="Jose Entregas"
        src=""
        style=""
      />
    </div>
  </div>
</header>
`;
